packages/hydra-processor/src/executor/TransactionalExecutor.ts

```diff
@@ -151,6 +152,15 @@ class EntityIdGenerator {
     this.lastKnownEntityId = nextEntityIdChars.join('')
     return this.lastKnownEntityId
   }
+
+  public async getNextEntityId(em: EntityManager): Promise<string> {
```

Comment:
Will this implementation work even in high concurrency. For example, let's say the function will be called five times before `this.nextEntityIdPromise` resolves - will everything execute properly or the first call will be ok, but the rest of them will only wait for the said first call to finish, and then unpredictable behavior will occur? 

In other words, don't we need a queue here? If this can at least theoretically happen, we should create a test for this, right?

Reply:
This shouldn't be an issue, because only the first call to `getNextEntityId` actually performs an asynchronous operation (querying the database).

Reply:
Actually, you were right, it did cause some issues.
I added a fix with `async-lock` and a test in https://github.com/Joystream/hydra/pull/474/commits/5f8450a2d8e10ae277e539ae8cc2e70c708ef299